src/pymmcore_plus/mda/metadata/schema.py

```diff
@@ -0,0 +1,169 @@
+from typing import Any, Dict, List, Literal, Optional, Tuple, TypedDict
+
+import useq
+from typing_extensions import NotRequired
+
+__all__ = [
+    "FrameMetaV1",
+    "SummaryMetaV1",
+    "ConfigGroup",
+    "ConfigPreset",
+    "DeviceInfo",
+    "ImageInfo",
+    "PixelSizeConfigPreset",
+    "Position",
+    "PropertyInfo",
+    "PropertyValue",
+    "SystemInfo",
+]
+
+
+class PropertyInfo(TypedDict):
+    """Information about a single device property."""
+
+    name: str
+    value: Optional[str]
+    data_type: Literal["undefined", "float", "int", "str"]
+    allowed_values: Optional[Tuple[str, ...]]
+    is_read_only: bool
+    is_pre_init: bool
+    limits: NotRequired[Tuple[float, float]]
+    sequenceable: bool
+    sequence_max_length: NotRequired[int]
+    # device_label: str
+
+
+class DeviceInfo(TypedDict):
+    """Information about a specific device."""
+
+    label: str
+    library: str
+    name: str
+    type: str
+    description: str
+    properties: Tuple[PropertyInfo, ...]
+    parent_label: Optional[str]  # None on hub devices
+
+    # state device only
+    labels: NotRequired[Tuple[str, ...]]
+    # focus device only
+    focus_direction: NotRequired[Literal["Unknown", "TowardSample", "AwayFromSample"]]
+    # hub device only
+    child_names: NotRequired[Tuple[str, ...]]
+    # stage or XYstage device only
+    is_stage_sequenceable: NotRequired[bool]
+    # stage device only
+    is_continuous_focus_drive: NotRequired[bool]
+    is_stage_linear_sequenceable: NotRequired[bool]
+    # camera device only
+    is_exposure_sequenceable: NotRequired[bool]
+
+
+class SystemInfo(TypedDict):
+    """General system information."""
+
+    pymmcore_version: str
+    pymmcore_plus_version: str
+    mmcore_version: str
+    device_api_version: str
+    device_adapter_search_paths: Tuple[str, ...]
+    system_configuration: Optional[str]
+    primary_log_file: str
+    circular_buffer_memory_footprint: int
+    buffer_total_capacity: int
+    buffer_free_capacity: int
```

Comment:
I would use `sequence_buffer_size` (rather than "footprint"), possibly with a unit suffix (MB or bytes). "Circular buffer" is (actually _was_) an implementation detail.

`buffer_total_capacity` is that size divided by the image size so is redundant, and also I'm not sure the CMMCore method returns a meaningful value before the buffer is (re)allocated for the current image size. Consider leaving out.

`buffer_free_capacity` should be left out because it is a transient state that has more to do with what you were doing previously, and I can't think of a situation where it would be useful.

Reply:
`sequence_buffer_size_mb` 👍

Reply:
done